src/mb_work_code_toolbox/index.ts

```diff
@@ -0,0 +1,786 @@
+        // eslint-disable-next-line no-unsanitized/method -- Fine.
+        logDiv.insertAdjacentHTML('beforeend', `
+            <div style="border: 1px dashed gray; padding: 2px 2px 5px 5px; margin-top: 2px; ${LOG_STYLES[level]}">${html}</div>`);
+        logDiv.style.display = 'block';
+        logDiv.scrollTop = logDiv.scrollHeight;
+    }
+
+    private addValidationError(agencyKey: string, code: string, message?: string): void {
+        const logDiv = this.form.querySelector<HTMLDivElement>(VALIDATION_LOG_QUERY)!;
+        let msg = `${code} does not look like a valid ${agencyKey}.`;
+        if (message) {
+            msg += ' ' + message;
+        }
+        // eslint-disable-next-line no-unsanitized/method -- Fine.
+        logDiv.insertAdjacentHTML('beforeend',
+            `<div style="border: 1px dashed gray; padding: 2px 2px 5px 5px; margin-top: 2px; ${LOG_STYLES['error']}">${msg}</div>`);
+        logDiv.style.display = 'block';
+        logDiv.scrollTop = logDiv.scrollHeight;
+    }
+
+    private addFormatWarning(agencyKey: string, code: string): void {
+        const logDiv = this.form.querySelector<HTMLDivElement>(VALIDATION_LOG_QUERY)!;
+        const msg = `${code} is not a well-formatted ${agencyKey}.`;
+        // eslint-disable-next-line no-unsanitized/method -- Fine.
+        logDiv.insertAdjacentHTML('beforeend',
+            `<div style="border: 1px dashed gray; padding: 2px 2px 5px 5px; margin-top: 2px; ${LOG_STYLES['warning']}">${msg}</div>`);
+        logDiv.style.display = 'block';
+        logDiv.scrollTop = logDiv.scrollHeight;
+    }
+}
+
+class WorkEditForm extends BaseWorkForm {
+    protected addToolsUI(): void {
+        // eslint-disable-next-line no-unsanitized/method -- Fine.
+        this.form.querySelector('.documentation')!.insertAdjacentHTML('beforebegin', mainUIHTML);
+    }
+}
+
+class IframeEditForm extends BaseWorkForm {
+    protected addToolsUI(): void {
+        // eslint-disable-next-line no-unsanitized/method -- Fine.
+        this.form.querySelector('.half-width')!.insertAdjacentHTML('beforebegin', mainUIHTML);
+        this.form.querySelector<HTMLElement>('#ropdebee-work-menu')!.style.marginLeft = '0px';
+    }
+}
+
+
+function editFormFactory(theForm: HTMLFormElement, inIframe: boolean): BaseWorkForm {
+    if (inIframe) {
+        return new IframeEditForm(theForm);
+    }
+
+    return new WorkEditForm(theForm);
+}
+
+function handleMB(): void {
+    const editWorkFormQuery = 'form.edit-work';
+
+    function handleChange(): void {
+        // Whenever a change occurs, try to add buttons to the attr table.
+        // FIXME: We should check the changes in more detail before querying.
+        const workForms: Array<[HTMLFormElement, boolean]> = [...document.querySelectorAll<HTMLFormElement>(editWorkFormQuery)].map((f) => [f, false]);
+        document.querySelectorAll('iframe').forEach((iframe) => {
+            if (!iframe.contentWindow) return;
+
+            onWindowLoaded(() => {
```

Comment:
How the hell did this ever work in the rewritten version? We're asynchronously adding stuff to an array but then consuming the array synchronously 🤦